src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1042,7 +1042,7 @@ private static object DeepCopierHelper(Type t, object original)
         /// </summary>
         /// <param name="t">The type.</param>
         /// <returns>true if <paramref name="t"/> is serializable, false otherwise.</returns>
-        internal static bool HasSerializer(Type t)
+        public static bool HasSerializer(Type t)
```

Comment:
I'm hesitant to make this member public, since miss-use could have bad side-effects, specifically since locking in app code is heavily discouraged. There are the Tester and TesterInternal projects (and TestInternalGrains & interfaces) specifically created to test these internal things. I believe it would be more appropriate to add all of the code in this PR in those internal projects, and keep this member as internal too.